docs/concepts/fundamentals/html_js_css_files.md

```diff
@@ -118,6 +118,92 @@ class Button(Component):
 </button>
 ```
 
+### Dynamic templates
```

Comment:
Mentioned how use `Component.on_render()` to customize what gets rendered - Basically covering what was asked here https://github.com/django-components/django-components/discussions/1204